doc/arch/014-reporting-etl-pipeline.md

```diff
@@ -0,0 +1,150 @@
+
+Once the reporting tables are populated, we will use Metabase to create and maintain the SQL queries that run against
+the reporting tables to generate reports.
+
+```
+ --------------------             ------------------------------------             ------------------
+| Application tables |    -->    | Reporting tables as database views |    -->    | Metabase reports |
+ --------------------             ------------------------------------             ------------------
+eg. blood_pressures              eg. blood_pressures_over_time                    eg. monthly control rate query
+                                 (latest BP for each patient as of every month)   (as of every month, how many BPs were
+                                                                                   controlled and recent)
+```
+
+The reporting tables will be implemented as database views within the Simple database.
+
+### Reporting Tables
+
+To start, the reporting tables will be granular de-normalized records of patients, blood pressures, encounters, and
+prescribed drugs.
+
+* Time: Each record will feature a month as dimension of time
+* Location: Each record will feature a facility (facility ID) as dimension of location
+
+For example, if "Ashish" is a patient in Simple, the "blood pressures over time" table will have records like:
+* As of Jan 2020, what was Ashish's most recent BP?
+* As of Feb 2020, what was Ashish's most recent BP?
+* As of Mar 2020, what was Ashish's most recent BP?
+* ...and so on
+
+The "prescription drugs over time" table will have records like:
+* As of Jan 2020, what drugs were prescribed to Ashish?
+* As of Feb 2020, what drugs were prescribed to Ashish?
+* As of Mar 2020, what drugs were prescribed to Ashish?
+* ...and so on
+
+Since these tables are pre-computing questions like "What was the most recent BP as of X month", the reports queries
+become much simpler. Instead of dealing with this question, the reports query can instead fetch all records for a
+particular month, and the "most recent" information will be readily available in the record.
+
+### Calendar Table
+
+Time, down to the month, is an important dimension of all reporting tables. As such, a calendar table will be created to
+support a consistent and robust way to describe months across all reporting tables. Calendar tables are a common
+occurrence in data warehouse design ([Here's one such example](https://www.mssqltips.com/sqlservertip/4054/creating-a-date-dimension-or-calendar-table-in-sql-server/)).
+
+The important elements of our calendar will be:
+* Cardinal month (1-12)
+* Year
+* Quarter: Quarterly analysis is common, so precomputing will be beneficial.
+* A unique date, like first of the month: This helps uniquely identify or group by each record in the calendar table on
+  a single column. `GROUP BY (year, month)` is more awkward.
+
+The calendar table will usually be joined with Simple's application tables to populate the denormalized reporting
```

Comment:
Will the calendar table assume that queries using it specify their time zone...I'm assuming UTC? Maybe a good thing to call out here.